src/ParallelAlgorithms/Interpolation/Protocols/InterpolationTargetTag.hpp

```diff
@@ -0,0 +1,123 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <string>
+#include <type_traits>
+
+#include "ParallelAlgorithms/Interpolation/InterpolationTargetDetail.hpp"
+#include "ParallelAlgorithms/Interpolation/Protocols/ComputeTargetPoints.hpp"
+#include "ParallelAlgorithms/Interpolation/Protocols/ComputeVarsToInterpolate.hpp"
+#include "ParallelAlgorithms/Interpolation/Protocols/PostInterpolationCallback.hpp"
+#include "Utilities/ProtocolHelpers.hpp"
+
+namespace intrp::protocols {
+namespace detail {
+// Both will have `::value = true` because we want the static assert in the
+// protocol to pass regardless of if the `compute_vars_to_interpolate` alias
+// exists if the conforming type. It's just that if the alias does exist,
+// there's an additional static assert to ensure that the alias conforms to the
+// ComputeVarsToInterpolate protocol
+template <typename ConformingType, bool HasAlias>
+struct if_alias_exists_assert_conforms_to;
+
+template <typename ConformingType>
+struct if_alias_exists_assert_conforms_to<ConformingType, false> {
+  static constexpr bool value = true;
+};
+
+template <typename ConformingType>
+struct if_alias_exists_assert_conforms_to<ConformingType, true> {
+  static constexpr bool value = true;
+  static_assert(tt::assert_conforms_to<
+                typename ConformingType::compute_vars_to_interpolate,
+                ComputeVarsToInterpolate>);
+};
+
+template <typename ConformingType, bool HasAlias>
+constexpr bool if_alias_exists_assert_conforms_to_v =
+    if_alias_exists_assert_conforms_to<ConformingType, HasAlias>::value;
+}  // namespace detail
+
+/*!
+ * \brief A protocol for `InterpolationTargetTag`s that are used in the
+ * intrp::InterpolationTarget parallel component.
+ *
+ * \details A struct conforming to the `InterpolationTarget` protocol must
+ * have
+ *
+ * - a type alias `temporal_id` that tells the interpolation target what values
+ *   of time to use.
+ *
+ * - a static function `name()` that returns a `std::string` used to uniquely
+ *   identify this InterpolationTargetTag
```

Comment:
You added this requirement in this PR, right? I suggest you make it optional and use `Options::name<>()` where needed (last commit), which falls back to the name of the struct. This avoids adding the `pretty_type` code that you added to every interpolation target. (we should probably move `Options::name` somewhere more general or merge it with `pretty_type`..).

Reply:
I have plans to do exactly that: replace `pretty_type::short_name` with `pretty_type::name` that does exactly what `Options::name` does. I'm getting tired of adding options to files that don't have anything to do with initialization just to use the `Options::name` function. For now I'd like to keep options out of the interpolation framework though (except where actually needed)

Reply:
`pretty_type::name` sounds good. For the moment though you only need to include the options header to use `Options::name`, no other option things necessary, so I think that's not too bad?